### DISCUSSION TITLE
v4.5.0

### DISCUSSION THREAD

**Discussion by AngledLuffa** [stanfordnlp/CoreNLP#1285] [Announcements]
# CoreNLP 4.5.0

Main features are improved lemmatization of English, improved tokenization of both English and non-English flex-based languages, and some updates to tregex, tsurgeon, and semgrex

- All PTB and German tokens normalized now in PTBLexer (previously only German umlauts).
This makes the tokenizer 2% slower, but should avoid issues with resume' for example 
  https://github.com/stanfordnlp/CoreNLP/commit/d46fecd93c6964f635efe85d9b7c327ee8880fb9

- log4j removed entirely from public CoreNLP (internal "research" branch still has a use) 
  https://github.com/stanfordnlp/CoreNLP/commit/f05cb54ec0a4f3c90395771817f44a81eb549baf

- Fix NumberFormatException showing up in NER models: https://github.com/stanfordnlp/CoreNLP/issues/547  https://github.com/stanfordnlp/CoreNLP/commit/5ee2c391104109a338a28f35c647b7684b00ad41

- Fix "seconds" in the lemmatizer: https://github.com/stanfordnlp/CoreNLP/commit/e7a073bde9ba7bbdb40ba81ed96d379455629e44

- Fix double escaping of & in the online demos: https://github.com/stanfordnlp/CoreNLP/commit/8413fa1fc432aa2a13cbb4a296352bb9bad4d0cb

- Report the cause of an error if "tregex" is asked for but no parse annotator is added: https://github.com/stanfordnlp/CoreNLP/commit/4db80c051322697c983ecda873d8d38f808cb96c

- Merge ssplit and cleanxml into the tokenize annotator (done in a backwards compatible manner): https://github.com/stanfordnlp/CoreNLP/pull/1259

- Custom tregex pattern, ROOT tregex pattern, and tsurgeon operation for simultaneously moving a subtree and pruning anything left behind, used for processing the Italian VIT treebank in stanza:   https://github.com/stanfordnlp/CoreNLP/pull/1263

- Refactor tokenization of punctuation, filenames, and other entities common to all languages, not just English: https://github.com/stanfordnlp/CoreNLP/commit/3c40ba32ca51af02936b907d03406e2158883f7b  https://github.com/stanfordnlp/CoreNLP/commit/58a2288239f631df47fac3eed105fe78c08b1a5d  https://github.com/stanfordnlp/CoreNLP/commit/8b97d64e48e6d4161f62a8635d2bb4cee2e95553

- Improved tokenization of number patterns, names with apostrophes such as Sh'reyan, non-American phone numbers, invisible commas    https://github.com/stanfordnlp/CoreNLP/commit/9476a8eb724e01df4b05bce38789dd8a7e61397c   https://github.com/stanfordnlp/CoreNLP/commit/6193934af8ae0abb0b4c6a2522d7efdfa426e5b3   https://github.com/stanfordnlp/CoreNLP/commit/afb1ea89c874acd58bab584f1e29a059c44dfd20  https://github.com/stanfordnlp/CoreNLP/commit/7c84960df4ac9d391ef37855572e2f8bc301ee17

- Significant lemmatizer improvements: adjectives & adverbs, along with some various other special cases  https://github.com/stanfordnlp/CoreNLP/pull/1266

- Include graph & semgrex indices in the results for a semgrex query (will make the results more usable)  https://github.com/stanfordnlp/CoreNLP/commit/45b47e245c367663bba2e81a26ea7c29262ad0d8

- Trim words in the NER training process.  spaces can still be inside a word, but random whitespace won't ruin the performance of the models  https://github.com/stanfordnlp/CoreNLP/commit/0d9e9c829bfa75bb661cccea03fc682a0f955f0d

<hr /><em>This discussion was created from the release <a href='https://github.com/stanfordnlp/CoreNLP/releases/tag/v4.5.0'>v4.5.0</a>.</em>

**Comment by paulk-asert** [comment]
Hi, congrats on the release! Is there a plan for publication to Maven central?

**Comment by AngledLuffa** [comment]
Hopefully in the next few days!  We were just making sure there are no
horrible bugs
